From 6008ee84d5599bc5fa55865b2ba0bb03e982c7b7 Mon Sep 17 00:00:00 2001
Message-Id: <6008ee84d5599bc5fa55865b2ba0bb03e982c7b7.1440053504.git.chang-joon.lee@intel.com>
From: Praveen Paneri <praveen.paneri@intel.com>
Date: Mon, 13 Jul 2015 14:19:28 +0530
Subject: [PATCH 1/5] SQUASHME! [VPG]: drm/i915: Avoid reading uninitialized
 dpst reg variable

If dpst init fails or not completed before using dpst status debugfs
function, system hangs due to accessing uninitialized register
variables. This patch fixes this by adding a check before accessing
dpst registers.

This commit can be merged in the following commit which originally
introduced reading from reg varibles.
  Author: Daisy Sun <daisy.sun@intel.com>
  Date:   Mon May 5 18:58:49 2014 -0700
  subjet: MUST_REBASE [VPG]: drm/i915: DPST debugfs read from reg varible
  Change-Id: I0eb236fe4ade4065c152ee42952f10e9a207cc77

Change-Id: I9682dfd35bcfe3e0658492dead4afa217cb20117
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13316
Signed-off-by: Praveen Paneri <praveen.paneri@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index e31f082..668e9ab 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -1839,6 +1839,12 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 		return 0;
 	}
 
+	/* Check if dpst is initialized by UMD? */
+	if (!dev_priv->dpst.reg.blm_hist_ctl) {
+		seq_puts(m, "DPST feature not enabled\n");
+		return 0;
+	}
+
 	mutex_lock(&dev_priv->dpst.ioctl_lock);
 
 	intel_display_power_get(dev_priv, PIPE_A);
-- 
1.9.1

